--- socialcalctableeditor.js.orig	2008-11-12 02:34:36.000000000 -0800
+++ socialcalctableeditor.js	2008-11-12 02:49:41.000000000 -0800
@@ -838,7 +838,7 @@ SocialCalc.ProcessEditorColsizeMouseDown
       editor.context.colwidth[mouseinfo.mouseresizecolnum] + '</div></td></tr></table>';
    SocialCalc.setStyles(sizedisplay.firstChild.lastChild.firstChild.childNodes[0], "filter:alpha(opacity=85);opacity:.85;"); // so no warning msg with Firefox about filter
 
-   editor.toplevel.appendChild(sizedisplay);
+   document.body.appendChild(sizedisplay);
 
    // Event code from JavaScript, Flanagan, 5th Edition, pg. 422
    if (document.addEventListener) { // DOM Level 2 -- Firefox, et al
@@ -936,7 +936,7 @@ SocialCalc.FinishColsize = function() {
    var editor = mouseinfo.editor;
    if (!editor) return;
 
-   editor.toplevel.removeChild(mouseinfo.mouseresizedisplay);
+   mouseinfo.mouseresizedisplay.parentNode.removeChild(mouseinfo.mouseresizedisplay);
    mouseinfo.mouseresizedisplay = null;
 
    editor.FitToEditTable();
@@ -2457,7 +2457,7 @@ SocialCalc.InputEcho = function(editor) 
 
    SocialCalc.DragRegister(this.main, true, true, null);
 
-   editor.toplevel.appendChild(this.main);
+   document.body.appendChild(this.main);
 
    }
 
@@ -2646,7 +2646,7 @@ SocialCalc.CreateTableControl = function
    if (scc.TCendcapClass) control.endcap.className = scc.TCendcapClass;
    AssignID(control.editor, control.endcap, "endcap"+vh);
 
-   control.main.appendChild(control.endcap);
+   document.body.appendChild(control.endcap);
 
    control.paneslider = document.createElement("div");
    s = control.paneslider.style;
@@ -2669,7 +2669,7 @@ SocialCalc.CreateTableControl = function
 
    SocialCalc.DragRegister(control.paneslider, control.vertical, !control.vertical, functions);
 
-   control.main.appendChild(control.paneslider);
+   document.body.appendChild(control.paneslider);
 
    control.lessbutton = document.createElement("div");
    s = control.lessbutton.style;
@@ -2692,7 +2692,7 @@ SocialCalc.CreateTableControl = function
 
    SocialCalc.ButtonRegister(control.lessbutton, params, functions);
 
-   control.main.appendChild(control.lessbutton);
+   document.body.appendChild(control.lessbutton);
 
    control.morebutton = document.createElement("div");
    s = control.morebutton.style;
@@ -2715,7 +2715,7 @@ SocialCalc.CreateTableControl = function
 
    SocialCalc.ButtonRegister(control.morebutton, params, functions);
 
-   control.main.appendChild(control.morebutton);
+  document.body.appendChild(control.morebutton);
 
    control.scrollarea = document.createElement("div");
    s = control.scrollarea.style;
@@ -2735,7 +2735,7 @@ SocialCalc.CreateTableControl = function
 
    SocialCalc.ButtonRegister(control.scrollarea, params, functions);
 
-   control.main.appendChild(control.scrollarea);
+   document.body.appendChild(control.scrollarea);
 
    control.thumb = document.createElement("div");
    s = control.thumb.style;
@@ -2760,7 +2760,7 @@ SocialCalc.CreateTableControl = function
              hoverstyle:  "backgroundImage:url("+imageprefix+"-thumb-"+vh+"h.gif)"};
    SocialCalc.ButtonRegister(control.thumb, params, null); // give it button-like visual behavior
 
-   control.main.appendChild(control.thumb);
+   document.body.appendChild(control.thumb);
 
    return control.main;
 
@@ -2907,7 +2907,7 @@ SocialCalc.TCPSDragFunctionStart = funct
       draginfo.trackingline.style.left = (editor.colpositions[col] || editor.headposition.left)+"px";
       }
 
-   editor.griddiv.appendChild(draginfo.trackingline);
+   document.body.appendChild(draginfo.trackingline);
 
    }
 
@@ -3000,7 +3000,7 @@ SocialCalc.TCPSDragFunctionStop = functi
    editor.FitToEditTable();
    editor.context.CalculateColWidthData();
 
-   editor.griddiv.removeChild(draginfo.trackingline);
+   draginfo.trackingline.parentNode.removeChild(draginfo.trackingline);
 
    editor.EditorRenderSheet();
 
@@ -3046,7 +3046,7 @@ SocialCalc.TCTDragFunctionStart = functi
       }
 
 
-   editor.toplevel.appendChild(draginfo.thumbstatus);
+   document.body.appendChild(draginfo.thumbstatus);
 
    }
 
@@ -3138,7 +3138,7 @@ SocialCalc.TCTDragFunctionStop = functio
    editor.FitToEditTable();
    editor.context.CalculateColWidthData();
 
-   editor.toplevel.removeChild(draginfo.thumbstatus);
+   draginfo.thumbstatus.parentNode.removeChild(draginfo.thumbstatus);
 
    editor.EditorRenderSheet();
 
